ert3/config/_stages_config.py

```diff
@@ -46,19 +46,28 @@ def _ensure_mime(cls: Type[_StagesConfig], field: str, values: Dict[str, Any]) -
         if not ert.serialization.has_serializer(guess):
             return DEFAULT_RECORD_MIME_TYPE
         return guess
-    return (
-        DEFAULT_CMD_MIME_TYPE
-        if cls == TransportableCommand
-        else DEFAULT_RECORD_MIME_TYPE
-    )
+    elif cls == TransportableCommand:
+        return DEFAULT_CMD_MIME_TYPE
+    return DEFAULT_RECORD_MIME_TYPE
+
+
+def _check_mime_is_dir(
+    cls: Type[_StagesConfig], values: Dict[str, Any]
+) -> Dict[str, Any]:
+    """If field is_directory is True, then set mime accordingly"""
+    if values["is_directory"]:
+        values["mime"] = "application/x-tar"
+    return values
 
 
 class Record(_StagesConfig):
     record: str
     location: str
     mime: str = ""
+    is_directory: bool = False
```

Comment:
What I don't understand fully here is that we have both `is_directory` which indicates that the record is a directory, but we also say that the `mime` is always `application/x-tar` in those cases.

Why use both `mime` and `is_directory` to convey this? IMO `is_directory` should trigger the `TarTransformation` and the `mime` should simply be `application/octet-stream`.

Or am I missing something here?

Reply:
> Edit: Is this because load_collection_from_file doesn't have is_directory?

Yes, that was the main reason. Will try without ...